src/weatherSource/WeatherSource.js

```diff
@@ -1,129 +1,90 @@
-import {
-    relative
-} from "path";
+//do własnego użytku udostępniam klucz api
+//API_KEY: 953349aaa2569f9cd4821f8c2ffda23a
 
-const API_KEY = '953349aaa2569f9cd4821f8c2ffda23a'
+// przykładowa inicjalizacja obiektu:
+// var WeatherAPI = new OpenWeatherMapApi('http://api.openweathermap.org/data/2.5/','953349aaa2569f9cd4821f8c2ffda23a')
 
-class WeatherAPI {
+class OpenWeatherMapApi {
 
-    static _throw(text) {
-        throw text;
+    constructor(api_url_address, api_key) {
```

Comment:
Tu tylko mała uwaga co do nazewnictwa, raczej trzymajmy się jednej konwencji nazewnictwa dla funkcji i zmiennych, wszędzie indziej piszesz w camelCase a tu jakby Ci się zatęskniło za Pythonem 😄

Reply:
Taa ogółem chciałem zastosować praktykę z Javy, ale Js mi nie pozwolił użyć tej samej nazwy parametrów co parametrów w klasie(pojawiał się potem błąd że brak takiej zmiennej), dlatego zrobiłem taki mały "unik". Suma sumarum zmieniłem to na zwykły camelcase. 

Tylko jak potem zrobić takie parametry, by programista korzystający z tej klasy wiedział, że te parametry mają być stałe ? Może użyć adnotacji @const ?

Reply:
Jeśli ustawiasz coś w konstruktorze i nie chcesz, żeby ktoś to zmieniał, to możesz po prostu posłużyć się konwencją _nazwaZmiennej i nie napisać do niej żadnego settera. Zasadniczo nie uniemożliwia to zmiany, ale "sens" prywatnej zmiennej (nawet, jeśli to kwestia konwencji, nie właściwość języka) jest taki, żeby jej z zewnątrz nikt nie tykał inaczej, niż dostarczonymi metodami (a jak gettera ani settera nie ma, to znaczy tyle, że ludzie z zewnątrz powinni się od tej zmiennej odczepić).

Są też chyba sposoby na uniemożliwienie modyfikacji zmiennej, jakoś na prototypach się to robi? Ale nie znam się na tym i to jest dziwne, plus masa roboty...